workshops/geometric_pattern/README.md

```diff
@@ -0,0 +1,538 @@
+
+```js
+function draw() {
+  var alternatingRow = false;
+  for (var y = 0; y <= height; y = y+CIRCLE_RAD) {
+    if (alternatingRow) {
+
+    } else {
+
+    }
+    for (var x = 0; x <= width; x = x+CIRCLE_DIA) {
+      ellipse(x, 0, CIRCLE_DIA, CIRCLE_DIA);
+    }
+    alternatingRow = !alternatingRow;
+  }
+}
+```
+
+Now we just need to make this shift. We'll do this by modifying the start `x` value in our inner for-loop. Currently, the start `x` value is always 0, which results in every row starting at the x-coordinate 0.
+
+We can create a variable (`startX`) within `draw()` and use our conditional to set the value of `startX` to 0 or `CIRCLE_RAD`, depending on if it's an alternating row or not. Don't forget to modify the inner for-loop so that the value of `x` starts at `startX`.
+
+```js
+function draw() {
+  var alternatingRow = false;
+  var startX;
+  for (var y = 0; y <= height; y = y+CIRCLE_RAD) {
+    if (alternatingRow) {
+      startX = CIRCLE_RAD;
+    } else {
+      startX = 0;
+    }
+    for (var x = startX; x <= width; x = x+CIRCLE_DIA) {
+      ellipse(x, 0, CIRCLE_DIA, CIRCLE_DIA);
+    }
+    alternatingRow = !alternatingRow;
+  }
+}
+```
+
+Save and check it out!
+
+### Reversing Direction
+
+So you may have noticed that it seems that our circles are overlapping on the wrong side. We wanted them to overlap each other on the top, but they are overlapping on the bottom. This is because they are being drawn from top to bottom, and the upper circles are being drawn over.
+
+We can fix this by swapping the start and end values of `y` in the outer for-loop and instead of incrementing the value of `y`, we'll decrement it. Don't forget to change the condition in the for-loop to be greater than or equal to 0 instead of less than or equal to.
+
+```js
+function draw() {
+  var alternatingRow = false;
+  var startX;
+  for (var y = height; y >= 0; y = y-CIRCLE_RAD) {
+    if (alternatingRow) {
+      startX = CIRCLE_RAD;
+    } else {
+      startX = 0;
+    }
+    for (var x = startX; x <= width; x = x+CIRCLE_DIA) {
+      ellipse(x, 0, CIRCLE_DIA, CIRCLE_DIA);
+    }
+    alternatingRow = !alternatingRow;
+  }
+}
+```
+
+Save to see the changes!
+
+## Part IV: Adding Color
+
+Black and white is nice, but how about some color? p5.js provides several helpful functions, including: [`color()`](http://p5js.org/reference/#p5/color), [`fill()`](http://p5js.org/reference/#p5/fill), and [`stroke()`](http://p5js.org/reference/#p5/stroke) that help modify the graphics.
+
+### Changing Fill
+
+One way to use `color()` is to provide 3 arguments; each corresponding to [red (R), green (G), and blue (B) values](https://en.wikipedia.org/wiki/RGB_color_model).
+
+Let's choose our color to be red. The R, G, and B values for a bright red are 255, 0, and 0, respectively. We can create this color with `color(255, 0, 0)`.
```

Comment:
How RGB works is actually the most interesting part of this workshop, cuz the workshop is largely customizable through the colors.